src/util/std_types.h

```diff
@@ -553,34 +520,28 @@ class struct_tag_typet:public tag_typet
   }
 };
 
-/*! \brief Cast a generic typet to a \ref struct_tag_typet
- *
- * This is an unchecked conversion. \a type must be known to be \ref
- * struct_tag_typet.
- *
- * \param type Source type
- * \return Object of type \ref struct_tag_typet
- *
- * \ingroup gr_std_types
-*/
+/// \brief Cast a typet to a \ref struct_tag_typet
+///
+/// This is an unchecked conversion. \a type must be known to be \ref
+/// struct_tag_typet. Will fail with a precondition violation if type
+/// doesn't match.
+///
+/// \param type Source type.
+/// \return Object of type \ref struct_tag_typet
 inline const struct_tag_typet &to_struct_tag_type(const typet &type)
 {
   PRECONDITION(type.id()==ID_struct_tag);
   return static_cast<const struct_tag_typet &>(type);
 }
 
-/*! \copydoc to_struct_tag_type(const typet &)
- * \ingroup gr_std_types
-*/
+/// \copydoc to_struct_tag_type(const typet &)
 inline struct_tag_typet &to_struct_tag_type(typet &type)
 {
   PRECONDITION(type.id()==ID_struct_tag);
   return static_cast<struct_tag_typet &>(type);
 }
 
-/*! \brief A union tag type
-*/
-
+/// A union tag type, i.e., a union type with an identifier
```

Comment:
With each of these `tag_typet` you might also like to refer to to the untagged type